docs/blockchain-rpc/hedera-rpc.md

```diff
@@ -0,0 +1,302 @@
+Most request that do not change network state can be performed against a [Mirror Node](https://docs.hedera.com/hedera/core-concepts/mirror-nodes) and for many use cases calling a mirror node endpoint is the recommended approach.
+
+### Parameters
+
+    1. `Object` - signMessage parameters
+      1.1 `signerAccountId` : `String` - Hedera account id in the format `<network>:<shard>.<realm>.<num>-<optional-checksum>`
+      1.2 `query` : `String` base64 encoded Query
+
+### Returns
+
+    1. `Object` - Result of Query to Hedera consensus node
+      1.1. `response` : `String` - base64 encoding of Hedera API response
+
+
+## hedera\_signMessage
+
+This method accepts a plain text string value as input. If approved by the user, the controller UTF-8 encodes this message prepended with "\x19Hedera Signed Message:\n" plus the length of the message and signs the resulting bytes in the same manner as HAPI transactions are signed. The resulting signature(s) are transmitted back to the user encoded in a SignatureMap structure. The pseudo code for computing the signature is as follows:
+
+```javascript
+
+<Ed25519 or ECDSA Key>.sign("\x19Hedera Signed Message:\n" + len(message) + message)
+```
+
+### Parameters
+
+    1. `Object` - signMessage parameters
+      1.1 `signerAccountId` : `String` - hedera account id in the format `<network>:<shard>.<realm>.<num>-<optional-checksum>`
+      1.2 `message` : `String`
+
+### Returns
+
+    1. `Object` - signMessage result
+      1.1 `signatureMap` : `String` - base64 encoded SignatureMap
+
+### Example
+
+#### Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "hedera_signMessage",
+  "params": {
+    "signerAccountId": "hedera:testnet:0.0.12345"
+    "message": "Co8BKowBCocBChoKDAjchKWmBhDW..."
+  }
+}
+```
+
+#### Result
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "signatureMap": "CAAQABjMrxoYABIGCAAQABgHGIDIr..."
+  }
+}
+```
+
+## hedera\_getNodeAddresses
+
+While constructing a transaction for transmission to a controller, a dApp needs to choose which Hedera Network node shall receive the transaction prior to signing (this is a requirement of the Hedera API Protocol). While a dApp can easily obtain a list of potential Hedera Nodes, a controller may not have an all-inclusive list nor a path to the node’s gRPC endpoint. The hedera_getNodeAddresses method allows a dApp to request a list of node wallet addresses known to the controller. The controller should only include nodes in this list that it is willing and able to submit transactions to at the time of the request.
+
+### Returns
+
+    1. `Object` - signMessage result
+      1.1 `nodes` : `Array` - an array of strings <shard>.<realm>.<num>-<optional-checksum> format identifying an Hedera Address
```

Comment:
```suggestion
      1.1 `nodes` : `Array` - an array of strings <shard>.<realm>.<num>-<optional-checksum> format each identifying a node by its Hedera Address
```
nits: is the checksum truly, fully optional? can one member in the array have the checksum and the others not? will it be accepted without a checksum but with the `-` divider?  i'm assuming these nits aren't worth picking, but just in case the answer is 'no' to any of those, maybe change the word `format` to `[validated format](https://docs.hedera.com/hedera/core-concepts/accounts#account-id)` or otherwise link to the important nits 😄

Reply:
the checksum is actually optional, here's the reference - https://hips.hedera.com/hip/hip-15#backwards-compatibility

though...the divider is only applicable with the checksum :)